rero_mef/agents/viaf/api.py

```diff
@@ -399,15 +399,19 @@ def delete(self, force=True, dbcommit=False, delindex=False):
             force=True, dbcommit=dbcommit, delindex=delindex)
         mef_actions = []
         for mef_record in mef_records:
-            # delete associated MEF record
-            mef_actions.append(f'Mark as deleted MEF: {mef_record.pid}')
-            for agent_record in agents_records:
+            mef_actions.append(f'MEF: {mef_record.pid}')
+            if len(agents_records):
+                mef_actions.append(
+                    f'{agents_records[0].name}: {agents_records[0].pid} '
+                    f'MEF: {mef_record.pid} {Action.UPDATE.value}'
+                )
+            for agent_record in agents_records[1:]:
                 mef_record.delete_ref(agent_record)
             mef_record.pop('viaf_pid', None)
-            mef_record.mark_as_deleted(dbcommit=True, reindex=True)
+            mef_record.update(data=mef_record, dbcommit=True, reindex=True)
         AgentMefRecord.flush_indexes()
         # recreate MEF records for agents
-        for agent_record in agents_records:
+        for agent_record in agents_records[1:]:
```

Comment:
for my info : why skipping first ?

Reply:
If I have IDREF, RERO, and GND in one MEF with VIAF, I will have one MEF for IDREF one for RERO and one for GND afterwards. If I skip the first one I have one MEF less to recreate.